src/megatron/energon/savable_loader.py

```diff
@@ -1012,31 +1029,31 @@ def save_state_global(
     def restore_state(
         self,
         state: Optional[Sequence[Optional[SavableDataLoaderState]]],
-        src_rank: Optional[int],
     ) -> None:
         """Deprecated. Use `restore_state_global` (or `restore_state_rank`) instead."""
 
-        return self.restore_state_global(state, src_rank)
+        return self.restore_state_global(state)
 
     def restore_state_global(
         self,
         state: Optional[Sequence[Optional[SavableDataLoaderState]]],
-        src_rank: Optional[int],
+        *,
+        src_rank: Optional[int] = None,
```

Comment:
So, this is the local rank? Should we also rename it to `src_dp_rank`? Now, this seems inconsistent with the save function?